src/platform_impl/redox/mod.rs

```diff
@@ -0,0 +1,1336 @@
+        orbclient::K_9 => Some(VirtualKeyCode::Key9),
+
+        orbclient::K_TICK => Some(VirtualKeyCode::Grave),
+        orbclient::K_MINUS => Some(VirtualKeyCode::Minus),
+        orbclient::K_EQUALS => Some(VirtualKeyCode::Equals),
+        orbclient::K_BACKSLASH => Some(VirtualKeyCode::Backslash),
+        orbclient::K_BRACE_OPEN => Some(VirtualKeyCode::LBracket),
+        orbclient::K_BRACE_CLOSE => Some(VirtualKeyCode::RBracket),
+        orbclient::K_SEMICOLON => Some(VirtualKeyCode::Semicolon),
+        orbclient::K_QUOTE => Some(VirtualKeyCode::Apostrophe),
+        orbclient::K_COMMA => Some(VirtualKeyCode::Comma),
+        orbclient::K_PERIOD => Some(VirtualKeyCode::Period),
+        orbclient::K_SLASH => Some(VirtualKeyCode::Slash),
+        orbclient::K_BKSP => Some(VirtualKeyCode::Back),
+        orbclient::K_SPACE => Some(VirtualKeyCode::Space),
+        orbclient::K_TAB => Some(VirtualKeyCode::Tab),
+        //orbclient::K_CAPS => Some(VirtualKeyCode::CAPS),
+        orbclient::K_LEFT_SHIFT => Some(VirtualKeyCode::LShift),
+        orbclient::K_RIGHT_SHIFT => Some(VirtualKeyCode::RShift),
+        orbclient::K_CTRL => Some(VirtualKeyCode::LControl),
+        orbclient::K_ALT => Some(VirtualKeyCode::LAlt),
+        orbclient::K_ENTER => Some(VirtualKeyCode::Return),
+        orbclient::K_ESC => Some(VirtualKeyCode::Escape),
+        orbclient::K_F1 => Some(VirtualKeyCode::F1),
+        orbclient::K_F2 => Some(VirtualKeyCode::F2),
+        orbclient::K_F3 => Some(VirtualKeyCode::F3),
+        orbclient::K_F4 => Some(VirtualKeyCode::F4),
+        orbclient::K_F5 => Some(VirtualKeyCode::F5),
+        orbclient::K_F6 => Some(VirtualKeyCode::F6),
+        orbclient::K_F7 => Some(VirtualKeyCode::F7),
+        orbclient::K_F8 => Some(VirtualKeyCode::F8),
+        orbclient::K_F9 => Some(VirtualKeyCode::F9),
+        orbclient::K_F10 => Some(VirtualKeyCode::F10),
+        orbclient::K_HOME => Some(VirtualKeyCode::Home),
+        orbclient::K_UP => Some(VirtualKeyCode::Up),
+        orbclient::K_PGUP => Some(VirtualKeyCode::PageUp),
+        orbclient::K_LEFT => Some(VirtualKeyCode::Left),
+        orbclient::K_RIGHT => Some(VirtualKeyCode::Right),
+        orbclient::K_END => Some(VirtualKeyCode::End),
+        orbclient::K_DOWN => Some(VirtualKeyCode::Down),
+        orbclient::K_PGDN => Some(VirtualKeyCode::PageDown),
+        orbclient::K_DEL => Some(VirtualKeyCode::Delete),
+        orbclient::K_F11 => Some(VirtualKeyCode::F11),
+        orbclient::K_F12 => Some(VirtualKeyCode::F12),
+
+        _ => None,
+    }
+}
+
+fn element_state(pressed: bool) -> event::ElementState {
+    if pressed {
+        event::ElementState::Pressed
+    } else {
+        event::ElementState::Released
+    }
+}
+
+#[derive(Default)]
+struct EventState {
+    lshift: bool,
+    rshift: bool,
+    lctrl: bool,
+    rctrl: bool,
+    lalt: bool,
+    ralt: bool,
+    llogo: bool,
+    rlogo: bool,
+    left: bool,
+    middle: bool,
+    right: bool,
+}
```

Comment:
Could you use bitset with the help of `bitflag` crate.

Reply:
Done